lib/gringotts/gateways/monei.ex

```diff
@@ -252,22 +242,18 @@ defmodule Gringotts.Gateways.Monei do
   The following session shows how one would process a payment in one-shot,
   without (pre) authorization.
 
-      iex> opts = [currency: "EUR"] # The default currency is EUR, and this is just for an example.
+      iex> amount = %{amount: 40, currency: "EUR"}
```

Comment:
I still don't understand why were we passing `opts  = [currency: "EUR"]` in the first place? It was available all along in the gateway configs and hence in the gateway files in the lib.

Thoughts: @oyeb @kartikjagdale @arjun289 @zeus999 ?

Reply:
@ashish173 in gateway configs, it was the `default_currency`, to fallback to this currency if someone doesn't specify currency.

Reply:
I had implemented just 1 optional param, `currency`. So it was the only thing that I could show an example for :smile:
I've always been against having `default_currency` in gateway/lib-global configs. Thankfully, with this PR all that will go away.

Reply:
@oyeb why u r against with default_currency? what problems you find in that?

Reply:
@pkrawat1  agree with @ashish173  here that user should be able to pass a default currency through options.

Reply:
@kartikjagdale I don't think there is a need for that, if the user is passing a currency in `Money` struct, then what is the use of of passing `default currency` in options. it should be handled by our application our application only.

Reply:
yes, now we don't have the need for the default currency.

Reply:
Okay, I've read all the comments above and for the sake of organisation I am listing some points which would benefit from having `default_currency`.

* In the application Instead of using `"USD"`strings in all the requests we can have a single source of truth for currency per gateway(assuming we have multiple gateways) or a global setting for `default_currency` this way we will be able to use that to create our request params for lib.

This lead to better organisation per gateway. Please note that we don't use this `config` directly in the library. We rely on what is send in the request.